### DISCUSSION TITLE
Copilot doesn't work well with Unity3d C# ... totally disaster and i'll paid for this... :-(

### DISCUSSION THREAD

**Discussion by MadMacMad** [community/community#81030] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

after digging deeper into copilot... because i'll paid for (me idiot)
it doesn't give much useful code nor suggestions that work...
to compare i'll used ChatGPT ... ChatGPT fixed my problem in no time...
Copilot only gave suggestions that where not really useful... totally dissapointed... a disaster 
it was not a complicated task ... only adapt a working script from Camera to RawImage...
ChatGPT did it for free and first suggestion worked already ...
Copilot gave me some lame answers and no working code... 
i'm really pissed off by all the hype about Copilot... 
i can not recommend to use it with Unity3d  C# script...
maybe with other code it works better but Unity C# is bad ... very bad ... worse than ChatGPT 3 ...

Hope someone of the Copilot team reads this and they will fix it soon ... 
Copilot + Unity3d C# = Not usable

**Comment by ayim** [comment]
Hi @MadMacMad - just to clarify, were we using the chat functionality to try to make this migration?
CC: @rheapatel

--
Aaron
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Comment by MadMacMad** [comment]
hmm...
don't understand the question???
"were we using the chat functionality to try to make this migration?" ???

sure what else should i'll use?
again C# Unity3d is not good with Copilot...
filling in some suggestions is not the deal...
but fix for a broken function etc... not working... 
worked the last 2 weeks on a Unity3d project... 
to be honest... GPT gave me better code than Copilot gave... 
it fixed also every problem with first answer... 
Copilot messed my code up if i'll accept the answer when looking ok.. 
again to be honest... after all the buzz... and hype ... it was not very useful in my case... 
but to me as Mac user... it's normal... people don't care anymore... nor Unity does about bug's... 
everybody does me me me... we are the best our is the best etc... but when solid problems pop up...
nobody cares... same with bug's ... what a strange world it has become

**Reply by ayim** [reply]
Trying to understand this better:
- If I'm hearing you right, you're using Unity3D in VSCode Copilot on Mac?
- It seems that the suggestions it gives you in grey text is not ideal
- If you use the "fix" prompt for a function, that doesn't seem to work either?

**Reply by MadMacMad** [reply]
hello,
well it works somehow... its not that it does't work technically...
but it's not very useful for Unity3d and C# ... 
if i'll ask to fix a problem... 
the suggestion is often bad... 
if i'll tell than my optimisation ...
Copilot say i'm right  ... so @ the end my code is better and optimised as copilot...
also i'm on Mac OS X ... that might be also a problem ... hence everything is today for Windows first ;-)
well it'*s ok now.... if you start to write a script it helps with some suggestions.. and i'll be faster with copilot
but to fix real code problems... it doesn't work in my case... GPT 4 did fix the problem...
and yes VS Code, Unity3d and Mac M3 Max ... work also in UE 5.3... because Unity sucks to meanwhile on Mac...
it's time that Apple gives us a GameEngine that it optimised for it's eco system... Unity was on the right path but then the bad people took over in 2018 and since then it all went down.... so sad to see it everywhere except UE... they focus also on Mac and care more than Unity and others... and it will pay back for them... and if others would do it too...
